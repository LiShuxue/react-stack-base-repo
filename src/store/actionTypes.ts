export const ACTION_TYPE_A: string = 'ACTION_TYPE_A';
export const ACTION_TYPE_B: string = 'ACTION_TYPE_B';
