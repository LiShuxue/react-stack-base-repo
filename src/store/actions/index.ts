import { actionCreatorA } from './actionCreatorA';
import { asyncActionCreatorB } from './actionCreatorB';

export default {
  actionCreatorA,
  asyncActionCreatorB
};
