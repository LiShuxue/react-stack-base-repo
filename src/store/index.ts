import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducer from './reducers';

// preload state object key must be same as reducer
const preloadState = {
  stateA: 'this is preload A',
  stateB: 'this is preload B'
};

// we not have to pass the preloadState, then it will use the default value in reducer as the state
const store = createStore(reducer, preloadState, applyMiddleware(thunk));

export default store;