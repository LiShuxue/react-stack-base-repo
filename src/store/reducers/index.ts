import { combineReducers } from 'redux';

import { handleActionA } from './moduleAReducer';
import { handleActionB } from './moduleBReducer';

// preload state object key must be same as reducer
const reducer = combineReducers({
  stateA: handleActionA,
  stateB: handleActionB
})

export default reducer;