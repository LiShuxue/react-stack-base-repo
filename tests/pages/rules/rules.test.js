import React from 'react';
import { render, fireEvent, wait } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import Rules from 'src/pages/rules';
import store from 'src/store';

beforeAll(() => {
  expect.extend({ toMatchDiffSnapshot });
});
test('should render Rules page correctly', () => {
  const testRenderer = renderer.create(
    <Provider store={store}>
      <Rules />
    </Provider>
  );
  const dom = testRenderer.toJSON();
  expect(dom).toMatchSnapshot();
});

test('should have "click this button to change the stateA" button', () => {
  const { getByText } = render(
    <Provider store={store}>
      <Rules />
    </Provider>
  );
  const element = getByText('click this button to change the stateA');
  expect(element).toBeInTheDocument();
});

test('should have "click this button to async change the stateB" button', () => {
  const { getByText } = render(
    <Provider store={store}>
      <Rules />
    </Provider>
  );
  const element = getByText('click this button to async change the stateB');
  expect(element).toBeInTheDocument();
});

test('click first button should change the stateA immediately', () => {
  const { getByText, asFragment } = render(
    <Provider store={store}>
      <Rules />
    </Provider>
  );
  const element = getByText('click this button to change the stateA');
  const firstUI = asFragment();
  fireEvent.click(element);
  const secondUI = asFragment();
  expect(firstUI).toMatchDiffSnapshot(secondUI);
});

test('click second button should change the stateB asynchronous"', async () => {
  const { getByText, asFragment } = render(
    <Provider store={store}>
      <Rules />
    </Provider>
  );
  const element = getByText('click this button to async change the stateB');
  const firstUI = asFragment();
  fireEvent.click(element);
  await wait(() => getByText(/async changed/i));
  const secondUI = asFragment();
  expect(firstUI).toMatchDiffSnapshot(secondUI);
});
