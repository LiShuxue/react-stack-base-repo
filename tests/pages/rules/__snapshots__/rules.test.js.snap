// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click first button should change the stateA immediately 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,11 +7,11 @@
      <button>
        click this button to change the stateA
      </button>
      <br />
      <h1>
-       stateA: this is preload A
+       stateA: changed
      </h1>
      <br />
      <button>
        click this button to async change the stateB
      </button>"
`;

exports[`click second button should change the stateB asynchronous" 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -15,9 +15,9 @@
      <button>
        click this button to async change the stateB
      </button>
      <br />
      <h1>
-       stateB: this is preload B
+       stateB: async changed
      </h1>
    </div>
  </DocumentFragment>"
`;

exports[`should render Rules page correctly 1`] = `
<div>
  <h1>
    Rules
  </h1>
  <hr />
  <button
    onClick={[Function]}
  >
    click this button to change the stateA
  </button>
  <br />
  <h1>
    stateA: 
    this is preload A
  </h1>
  <br />
  <button
    onClick={[Function]}
  >
    click this button to async change the stateB
  </button>
  <br />
  <h1>
    stateB: 
    this is preload B
  </h1>
</div>
`;
